mysql-test/lib/mtr_report.pm

```diff
@@ -497,29 +497,40 @@ sub mtr_report_stats ($$$$) {
       $test_time = sprintf("%.3f", $test->{timer} / 1000);
       $test->{'name'} =~ s/$current_suite\.//;
 
-      my $test_result;
+      my $test_result= $test->{'result'};
 
-      # if a test case has to be retried it should have the result MTR_RES_FAILED in jUnit XML
       if ($test->{'retries'} > 0) {
-        $test_result = "MTR_RES_FAILED";
+        if ($test->{'result'} eq "MTR_RES_PASSED"){
+          $test_result = "MTR_RES_RETRY_PASSED";
+        }
+        else {
+          if ($test->{'failures'} > 1){
+            $test_result = "MTR_RES_RETRY_FAILED"; # retry failure
+          }
+        }
+      }
+
+      my $combinations;
+      if (defined($test->{combinations})){
+        $combinations = join ',', @{$test->{combinations}};
```

Comment:
could you, please, check whether the list of combinations is sorted at this point?
We want a stable list of combinations in all reports, so it is definitely sorted somehow somewhere.

But if not here, then you'll want `join ',', sort @{$test->{combinations}}`

Reply:
@DmitriyKarpovskiy1 can you please check this ^

Reply:
Initially, the combinations were not sorted. Added sorting.